ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/agent/MLAgentExecutor.java

```diff
@@ -222,6 +222,12 @@ public void execute(Input input, ActionListener<Output> listener) {
                                         String appType = mlAgent.getAppType();
                                         String question = inputDataSet.getParameters().get(QUESTION);
 
+                                        if (parentInteractionId != null && regenerateInteractionId != null) {
+                                            throw new IllegalArgumentException(
+                                                "Provide either `parent_interaction_id` to update an existing interaction, or `regenerate_interaction_id` to create a new one."
```

Comment:
Are you showing this error to customer? Does customer know what is `parent_interaction_id` or `regenerate_interaction_id` to provide?